src/application/clean.go

```diff
@@ -7,26 +7,29 @@ import (
 
 	"github.com/Originate/exosphere/src/docker/compose"
 	"github.com/Originate/exosphere/src/docker/composebuilder"
+	"github.com/Originate/exosphere/src/types"
 	"github.com/Originate/exosphere/src/util"
 )
 
-// CleanApplicationContainers cleans application Docker containers under appDir/docker-compose
-func CleanApplicationContainers(appDir, composeProjectName string, writer io.Writer) error {
-	dockerComposeFileNames := composebuilder.GetLocalRunComposeFileNames()
-	for _, dockerComposeFileName := range dockerComposeFileNames {
-		err := killIfExists(appDir, dockerComposeFileName, composeProjectName, writer)
+// CleanContainers cleans all cantainers listed in the yaml files under appDir/docker-compose
+func CleanContainers(appContext types.AppContext, writer io.Writer) error {
+	err := GenerateComposeFiles(appContext)
+	if err != nil {
+		return err
+	}
+	composeProjectName := composebuilder.GetDockerComposeProjectName(appContext.Config.Name)
+	for _, dockerComposeFileName := range composebuilder.GetComposeFileNames() {
+		if dockerComposeFileName == composebuilder.LocalTestComposeFileName {
+			composeProjectName = composebuilder.GetTestDockerComposeProjectName(appContext.Config.Name)
+		}
```

Comment:
I don't like changing the variable in the loop as it effects the following loop iterations. I think composeProjectName should be recomputed on each loop iteration.